### PR TITLE
nixos/cntlm: add types

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
#76184

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — DISMISSED** [review]
Please change the title to ``nixos/cntlm: add types``

**Review by cole-h — CHANGES_REQUESTED** [review]
Please rename your commit message to follow the format mentioned in our [CONTRIBUTING document](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) -- specifically, replace the backslash with a forward slash: `nixos/cntlm`.

Amongst other things, this makes it easier to grep the git log.

File: nixos/modules/services/networking/cntlm.nix [hunk 1]
```diff
@@ -75,11 +78,13 @@ in
         A list of domains where the proxy is skipped.
       '';
       default = [];
+      type = types.listOf types.str;
       example = [ "*.example.com" "example.com" ];
     };
 
     port = mkOption {
       default = [3128];
+      type = types.int;
```
Comment (lourkeur) [inline nixos/modules/services/networking/cntlm.nix:87]:
```suggestion
      type = types.listOf types.port;
```
1. looks like it needs to be a list
2. `types.port` exists and is more specific

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
~~You need to use a forward slash, not a backslash in the commit message.~~ Sorry I'm late on that.

Looks good otherwise.

**Comment by Scriptkiddi** [comment]
done

File: nixos/modules/services/networking/cntlm.nix [hunk 2]
```diff
@@ -75,11 +78,13 @@ in
         A list of domains where the proxy is skipped.
       '';
       default = [];
+      type = types.listOf types.str;
       example = [ "*.example.com" "example.com" ];
     };
 
     port = mkOption {
       default = [3128];
+      type = types.listOF types.port;
```
Comment (cole-h) [inline nixos/modules/services/networking/cntlm.nix:87]:
```suggestion
      type = types.listOf types.port;
```

**Review by cole-h — APPROVED** [review]
Diff LGTM, thanks.

**Pull request merged**